crates/proof-of-sql-parser/src/error.rs

```diff
@@ -14,4 +15,43 @@ pub enum ParseError {
     ResourceIdParseError(String),
 }
 
+/// TODO: Add docs
 pub type ParseResult<T> = std::result::Result<T, ParseError>;
+
+/// Errors related to time operations, including timezone and timestamp conversions.s
+#[derive(Error, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum PoSQLTimestampError {
+    /// Error when the timezone string provided cannot be parsed into a valid timezone.
+    #[error("invalid timezone string: {0}")]
+    InvalidTimezone(String),
+
+    /// Error indicating an invalid timezone offset was provided.
+    #[error("invalid timezone offset")]
+    InvalidTimezoneOffset,
+
+    /// Indicates a failure to convert between different representations of time units.
+    #[error("Invalid time unit")]
+    InvalidTimeUnit(String),
+
+    /// The local time does not exist because there is a gap in the local time.
+    /// This variant may also be returned if there was an error while resolving the local time,
+    /// caused by for example missing time zone data files, an error in an OS API, or overflow.
+    #[error("Local time does not exist because there is a gap in the local time")]
+    LocalTimeDoesNotExist,
+
+    /// The local time is ambiguous because there is a fold in the local time.
+    /// This variant contains the two possible results, in the order (earliest, latest).
+    #[error("Unix timestamp is ambiguous because there is a fold in the local time.")]
+    Ambiguous,
```

Comment:
Mapping `LocalResult::Ambiguous` to this basically loses information. It's probably better to either include that error or its stringified version.